### ISSUE TITLE
Encoding handling by corpus() function.

### ISSUE THREAD

**Issue by tpaskhalis** [kbenoit/quanteda#104]
There currently seem to be no checks on what is the encoding of passed texts to the `corpus()` function. I noticed some commented-out legacy code, where encoding was allowed to be passed explicitly as an argument, but no warnings that would let the user know that something is wrong with input texts.

This creates a problem with some in-built datasets, such as `ukManifestos`, which contains Latin-1 characters in the text `UK_natl_1997_en_SEP`. But it almost certainly is a more widespread issue with user-defined corpora. 

```
> summary(ukManifestos)
Corpus consisting of 101 documents, showing 100 documents.

 Error in stringi::stri_detect_charclass(x, "[A-Z]") : 
  invalid UTF-8 byte sequence detected. perhaps you should try calling stri_enc_toutf8() 
```

I'd suggest we have some try-catch error handling mechanism at the level of `corpus()` function. But for now I just added this functionality inside `nsentence()` to resolve the immediate issue. Probably, these Latin-1 characters should also be cleared from the `ukManifestos` corpus. They mostly consist of whitespace and newline characters in Latin-1 encoding as far as I can tell.

**Comment by kbenoit** [comment]
Agree with you on converting the Latin-1 characters from the ukManifestos corpus. 

However the issue with setting encoding for corpus texts is a complicated one, and I've tried several solutions that just don't seem to work cross-platform. Initially I wanted to convert all texts to UTF-8 internally, but Windows platform users were having problems as you apparently cannot set the R locale encoding to UTF-8 in Windows. So the idea was to import texts with conversion to the system encoding, to guarantee compatibility.

The `nsentence()` changes you made have now been incorporated, at least, so there is no error when it checks the case of the texts in `summary.corpus()`.

**Comment by kbenoit** [comment]
I think this is solved now, with the internal conversion into UTF-8 for all texts. R basically converts its text into either the native encoding, or if this cannot handle the characters, into UTF-8. I have now set `textfile()` to allow an input encoding but for the connection to translate this always into UTF-8, and to set the encoding bit to "UTF-8" (although this will not "stick" if only the lower 128 characters are used.

See `?encodedTextFiles` for a comprehensive set of tests using differently encoded texts. I have verified that this works in Windows and Linux, even when the Windows locale is set to 1252.
